rfd/0080-piv-private-key.md

```diff
@@ -0,0 +1,390 @@
+    private_key_policy: none | hardware_key | hardware_key_touch
+```
+
+- `none` (default): No enforcement on private key usage.
+- `hardware_key`: A user's private keys must be generated on a hardware key. As a result, the user cannot use their signed certificates unless they have their hardware key connected.
+- `hardware_key_touch`: A user's private keys must be generated on a hardware key, and must require touch to be accessed. As a result, the user must touch their hardware key on login, and on subsequent request (touch is cached on the hardware key for 15 seconds).
+
+#### Per-session MFA configuration
+
+```yaml
+auth_service:
+  ...
+  authentication:
+    ...
+    require_session_mfa: off | on | hardware_key | hardware_key_touch
+```
+```yaml
+kind: cluster_auth_preference
+version: v2
+metadata:
+  name: cluster-auth-preference
+spec:
+  require_session_mfa: off | on | hardware_key | hardware_key_touch
+```
+```yaml
+kind: role
+version: v5
+metadata:
+  name: role-name
+spec:
+  role_options:
+    require_session_mfa: off | on | hardware_key | hardware_key_touch
+```
+
+- `on`: Current per-session MFA functionality - users are required to pass an MFA challenge with a registered MFA device in order to start new SSH|Kubernetes|DB|Desktop sessions. Non-session requests, and app-session requests are not impacted.
+- `hardware_key`: Essentially combines `require_session_mfa: on` with `private_key_policy: hardware_key`. MFA tap is only required for session requests, but all Teleport API requests require certificates backed by a hardware private key.
+- `hardware_key_touch`: This is the same as `private_key_policy: hardware_key_touch`. Per-session MFA is disabled in favor of touch checks directly on the hardware key the user used to login.
+  - This will require some changes to the per-session MFA checks. If a user has `require_session_mfa: hardware_key_touch`, then this should override any other per-session MFA requirement, including the cluster-wide requirement or node/access specific requirements.
+
+This change will require changing the `require_session_mfa` fields above from a `bool` to a `string`. This will be handled by introducing a new proto field and custom marshalling logic to maintain interoperability between new and old servers and clients. See https://github.com/gravitational/teleport/pull/12054 for an example of this.
+
+Note: From a product perspective, we have decided to add the new `require_session_mfa` options and omit the new `private_key_policy` option to reduce configuration knobs. Since the underlying implementation will be separate from per-session MFA, this RFD will continue to refer to `private_key_policy` to make reasoning about the proposed changes easier to understand.
+
+#### Enforcement - Attestation
+
+In order to enforce user private key usage, we need to take a certificate's public key and tie it back to a trusted hardware device, which can be done with attestation, as explained [above](#attestation).
```

Comment:
While the current design seems to use the default management key (no problem with that), attestation won't be possibile if we decide to allow keys management/import. Unless product design strongly pushes for that, I would stick to trusted hardware keys generation only

Reply:
Agreed, the plan is to only use hardware-generated keys.